notifications.bs

```diff
@@ -905,19 +964,11 @@ attribute's getter must return the result of the following steps:
     <li><p>Set <var>action</var>'s {{NotificationAction/icon}} to
     <var>entry</var>'s <a>action icon URL</a>.
 
-    <!-- XXX IDL dictionaries are usually returned by value, so don't need to be
-         immutable. But FrozenArray reifies the dictionaries to mutable JS
-         objects accessed by reference, so we explicitly freeze them.
-         It would be better to do this with IDL primitives instead of JS - see
-         https://www.w3.org/Bugs/Public/show_bug.cgi?id=29004 -->
```

Comment:
So is the resource chosen synchronously so that it can be immediately reflected through a single `icon` attribute? Doesn't that mean that the bits implementing the notification API need to know quite a bit about the UI of the platform its running on?

AFACIT, the thing that was frozen here wasn't the value returned by the attribute (which is readonly), but the `NotificationAction` objects in that array. Otherwise one can do things like `notification.actions[0].icon = 'newicon.png'` which would have no effect. Exact same problem as discussed for Media Session artwork, in other words.